### PR TITLE
feat(opensearchservice): configuring gp3 throughput 

### LINKED ISSUE #26137: opensearch: add throughput to EbsOptions
### Describe the feature

The `interface EbsOptions` for the `opensearchservices` CDK construct is missing a provisioned `throughput` option for eg gp3 instance types.

`iops` is there, but not `throughput` (which can be configured in the web console.

### Use Case

To configure provisioned `throughput` in CDK.

### Proposed Solution

_No response_

### Other Information

_No response_

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.83.1

### Environment details (OS name and version, etc.)

macOS

### PR DESCRIPTION
**Pull request by sktan**
The interface EbsOptions for the opensearchservices CDK construct is missing a provisioned throughput option for eg gp3 instance types.
iops is there, but not throughput

Closes #26137.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-opensearchservice/lib/domain.ts [hunk 1]
```diff
@@ -1575,6 +1583,7 @@ export class Domain extends DomainBase implements IDomain, ec2.IConnectable {
         volumeSize: ebsEnabled ? volumeSize : undefined,
         volumeType: ebsEnabled ? volumeType : undefined,
         iops: ebsEnabled ? props.ebs?.iops : undefined,
+        throughput: ebsEnabled ? props.ebs?.throughput : undefined,
```
Comment (corymhall) [inline packages/aws-cdk-lib/aws-opensearchservice/lib/domain.ts:1586]: It doesn't look like we have any check on the volumeType. It says iops and
throughput are only supported for iops/gp3 types, but we only check if it's ebs
type?

Do you know what happens if `throughput` is defined for `gp2` volumes? Does it
just ignore that property or does it throw an error?

I think we need something like what we have in the `volume` construct
https://github.com/aws/aws-cdk/blob/53d24f1a1357410746052bbf25b5b89b985f7b80/packages/aws-cdk-lib/aws-ec2/lib/volume.ts#L704-L771

**Review by corymhall — CHANGES_REQUESTED** [review]

File: packages/aws-cdk-lib/aws-opensearchservice/lib/domain.ts (hunk 1 above)
Comment (sktan) [reply packages/aws-cdk-lib/aws-opensearchservice/lib/domain.ts:1586]: I did think about checks specification checks, the other option for `iops` didn't any checks defined so I didn't it for the `throughput` option`. I'll get that sorted in the next commit :)

Comment (sktan) [reply packages/aws-cdk-lib/aws-opensearchservice/lib/domain.ts:1668]: Added a bunch of tests for you @corymhall

**Comment by jumic** [comment]
@sktan The last build failed. Do you need help?
If the build is successful locally and only failed in this PR: Merge branch main into your branch to trigger a new build.

**Comment by sktan** [comment]
> @sktan The last build failed. Do you need help? If the build is successful locally and only failed in this PR: Merge branch main into your branch to trigger a new build.

Hey, thanks for the offer @jumic. I haven't gotten around to fixing it just yet, it failed due to another PR making a change to the default parameters causing my integration test to provide a `CHANGED` result.

This will get sorted soon

**Comment by sktan** [comment]
@corymhall we should be good to go again for this PR

File: packages/aws-cdk-lib/aws-opensearchservice/lib/domain.ts [hunk 2]
```diff
@@ -1468,6 +1476,88 @@ export class Domain extends DomainBase implements IDomain, ec2.IConnectable {
       throw new Error('EBS volumes are required when using instance types other than r3, i3 or r6gd.');
     }
 
+    // Only for a valid ebs volume configuration, per
+    // https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-opensearchservice-domain-ebsoptions.html
+    if (ebsEnabled) {
+      // Check if iops or throughput if general purpose is configured
+      if (volumeType == ec2.EbsDeviceVolumeType.GENERAL_PURPOSE_SSD || volumeType == ec2.EbsDeviceVolumeType.STANDARD) {
```
Comment (jogold) [inline packages/aws-cdk-lib/aws-opensearchservice/lib/domain.ts:1483]:
```suggestion
      if (volumeType === ec2.EbsDeviceVolumeType.GENERAL_PURPOSE_SSD || volumeType === ec2.EbsDeviceVolumeType.STANDARD) {
```

**Review by rix0rrr — DISMISSED** [review]

**Pull request merged**